lib/std/mem.zig

```diff
@@ -294,6 +294,14 @@ pub const Allocator = struct {
         new_buf[m.len] = 0;
         return new_buf[0..m.len :0];
     }
+
+    /// Returns a pointer to allocated memory with the provided value.
+    /// Call `destroy` with the result to free the memory.
+    pub fn dupeOne(allocator: *Allocator, value: var) Error!*@TypeOf(value) {
```

Comment:
This API is problematic because the purpose is to be convenient, but whether the `value` parameter is a pointer or the child type is unclear unless one reads the implementation, and it would be likely for the API user to pass the wrong one, potentially even causing a runtime bug in the code.

I agree that if you need a utility function like this in your codebase, it is pretty useful. But I also think that having this in the standard library could easily be considered a footgun, especially when you consider that for many types, simply copying the memory bytes would not be an appropriate way to do it. For example if you did `dupeOne` with `std.AutoHashMap` or `std.ArrayList` it would be a footgun.

I think having the user provide this function for themselves if they need it is just the right nudge to prevent many bugs from occurring in practice.

Reply:
We could call it e.g. dupeValue and make it clear from the *name* how it behaves?

Reply:
Feel free to either tweak this or close it as you like, it's fine with me regardless.